File: app/src/main/java/hsk/practice/myvoca/AppHelper.java
package hsk.practice.myvoca;

import android.Manifest;
import android.content.Context;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Calendar;

import database.source.VocaRepository;
import database.source.local.VocaDatabase;
import hsk.practice.myvoca.ui.activity.MainActivity;

public class AppHelper {

    private static ArrayList<String> permissionsList;

    public static ArrayList<String> getPermissionList() {
        if (permissionsList == null) {
            permissionsList = new ArrayList<>();
            permissionsList.add(Manifest.permission.READ_EXTERNAL_STORAGE);
            permissionsList.add(Manifest.permission.WRITE_EXTERNAL_STORAGE);
        }
        return permissionsList;
    }

    public static void writeLog(String text) {
        File logFile = new File("sdcard/MyVoca/log.txt");
        if (!logFile.exists()) {
            try {
                logFile.createNewFile();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        String timeString = getTimeString(System.currentTimeMillis());
        try {
            BufferedWriter bw = new BufferedWriter(new FileWriter(logFile, true));
            bw.append(timeString + ": " + text);
            bw.newLine();
            bw.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static String getTimeString(long timeInMilis) {
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(timeInMilis);

        int year = cal.get(Calendar.YEAR);
        int mon = cal.get(Calendar.MONTH);
        int day = cal.get(Calendar.DAY_OF_MONTH);
        int hour = cal.get(Calendar.HOUR_OF_DAY);
        int min = cal.get(Calendar.MINUTE);
        int sec = cal.get(Calendar.SECOND);
        return String.format("%d.%02d.%02d. %02d:%02d:%02d", year, mon + 1, day, hour, min, sec);
    }

    public static boolean isForeground(Context context) {
        return MainActivity.isRunning();
    }

    public static void loadInstance(Context context) {
        VocaDatabase.loadInstance(context);
        VocaRepository.loadInstance();
    }

    public static boolean isStringOnlyAlphabet(String str) {
        if (str == null || str.equals("")) {
            return false;
        }
        for (char c : str.toCharArray()) {
            if (!('a' <= c && c <= 'z') && !('A' <= c && c <= 'Z') && c != '%') {
                return false;
            }
        }
        return true;
    }
}

File: app/src/main/java/hsk/practice/myvoca/AppHelper.kt
package hsk.practice.myvoca

import android.Manifest
import hsk.practice.myvoca.ui.activity.MainActivity
import java.io.BufferedWriter
import java.io.File
import java.io.FileWriter
import java.io.IOException
import java.util.*

/**
 * Application helper class. Common methods are defined here.
 * All methods are static.
 */
object AppHelper {

    const val LOG_TAG = "HSK APP"

    private var permissionsList: ArrayList<String?>? = null

    /**
     * Returns a permission list of the application.
     *
     * @return permission list of the application
     */
    fun getPermissionList(): ArrayList<String?>? {
        if (permissionsList == null) {
            permissionsList = ArrayList()
            permissionsList!!.add(Manifest.permission.READ_EXTERNAL_STORAGE)
            permissionsList!!.add(Manifest.permission.WRITE_EXTERNAL_STORAGE)
        }
        return permissionsList
    }

    /**
     * Write a log to the file.
     * Not implemented completely, will be used at the future
     *
     * @param text string to write to a file
     */
    fun writeLog(text: String?) {
        val logFile = File("sdcard/MyVoca/log.txt")
        if (!logFile.exists()) {
            try {
                logFile.createNewFile()
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
        val timeString = System.currentTimeMillis().getTimeString()
        try {
            val bw = BufferedWriter(FileWriter(logFile, true))
            bw.append("$timeString: $text")
            bw.newLine()
            bw.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    /**
     * Check if the application is on the foreground.
     *
     * @return if application is on the foreground
     */
    fun isForeground(): Boolean {
        return MainActivity.isRunning()
    }
}